src/components/Setup/ProtocolCard.js

```diff
@@ -1,25 +1,125 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-
+import { connect } from 'react-redux';
+import cx from 'classnames';
+import { actionCreators as dialogActions } from '../../ducks/modules/dialogs';
+import { actionCreators as installedProtocolsActions } from '../../ducks/modules/installedProtocols';
 import { Icon } from '../../ui/components';
+import { APP_SUPPORTED_SCHEMA_VERSIONS, APP_SCHEMA_VERSION } from '../../config';
+
+const ProtocolCard = ({ protocol, openDialog, deleteProtocol, selectProtocol }) => {
```

Comment:
I still think this should be a class. Creating handlers etc in the rendering method causes React to update the dom every render, regardless of whether it has changed.

Reply:
> I still think this should be a class. Creating handlers etc in the rendering method causes React to update the dom every render, regardless of whether it has changed.

No sure, I just don't think that's a big deal with this component. The functions within render are trivial logic evaluations against an object.